lib/encoding/bufferreader.js

```diff
@@ -17,8 +17,12 @@ var BufferReader = function BufferReader(buf) {
       buf: buf
     });
   } else if (_.isString(buf)) {
+    var b = new Buffer(buf, 'hex');
+    if (b.length * 2 != buf.length) 
+      throw new TypeError('Invalid hex string');
```

Comment:
This is because Node 8 does not longer throw this error, and instead cut the input string:

ie: new Buffer('abc','hex') will result on ['ab'], with no error.